### PR TITLE
Registration Date Report Addition

### PR DESCRIPTION
**Pull request by Sheetal-VB**
- Addition of registration date based report
- Headers changes for Call reports
- Telugu Certificate addition

### CODE REVIEW

File: NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/dao/impl/BeneficiaryWithRegistrationDateBlockDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+                if(!set.contains(block.getBlockId().longValue() )){
+                    KilkariSubscriberRegistrationDateRejectedCountDto dto = new KilkariSubscriberRegistrationDateRejectedCountDto();
+                    dto.setLocationId(block.getBlockId());
+                    dto.setSubscriberCount(0);
+                    kilkariSubscriberRegistrationDateRejectedCountDtos.add(dto);
+                }
+            }
+        }
+        catch (Exception e){
+            LOGGER.info("Exception " + e);
```
Comment (rkjangid07) [inline NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/dao/impl/BeneficiaryWithRegistrationDateBlockDaoImpl.java]: log the exception in proper format

File: NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/dao/impl/BeneficiaryWithRegistrationDateBlockDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        }
+
```
Comment (rkjangid07) [inline NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/dao/impl/BeneficiaryWithRegistrationDateBlockDaoImpl.java]: If you add completion log

File: NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/business/impl/AggregateKilkariReportsServiceImpl.java [hunk 3]
```diff
@@ -521,6 +538,149 @@ private List<KilkariSubscriber> getKilkariSubscriberCount(Integer locationId, St
...
+                        LOGGER.debug("Adding record for " + kilkariSubscriberRegistrationDateListDto.getLocationName());
+                        kilkariSubscriberRegistrationDateListDtos.add(kilkariSubscriberRegistrationDateListDto);
+                    }
+                }
+
+            }
+        }
+        aggregateKilkariReportsDto.setTableData(kilkariSubscriberRegistrationDateListDtos);
+        LOGGER.info("Operation = getKilkariSubscriberCountReportBasedOnRegistrationDate , status = {} Report Data generated " , "COMPLETED");
+        return  aggregateKilkariReportsDto;
```
Comment (rkjangid07) [inline NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/business/impl/AggregateKilkariReportsServiceImpl.java]: COMPLETED is an constant

File: NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/dao/impl/BeneficiaryWithRegistrationDateBlockDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+                }
+                else{
+                    dto.setLocationId(((BigInteger) row[0]).longValue());
+                    dto.setLocationType("BLOCK");
+                }
+                kilkariSubscriberRegistrationDateDtoList.add(dto);
+            }
+        }
+        catch (Exception e){
+            LOGGER.info("Exception " + e);
```
Comment (rkjangid07) [inline NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/dao/impl/BeneficiaryWithRegistrationDateBlockDaoImpl.java]: add logs

File: NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/dao/impl/BeneficiaryWithRegistrationDateBlockDaoImpl.java (hunk 2 above)
Comment (Sheetal-VB) [reply NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/dao/impl/BeneficiaryWithRegistrationDateBlockDaoImpl.java]: Done

File: NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/business/impl/AggregateKilkariReportsServiceImpl.java (hunk 3 above)
Comment (Sheetal-VB) [reply NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/business/impl/AggregateKilkariReportsServiceImpl.java]: Done

**Pull request merged**